File: task-03/sub-task 4/Javascript/code.js
const fs = require('fs');

const n = parseInt(fs.readFileSync('/home/abinav/git/amfoss-tasks/task-03/sub-task 4/input.txt', 'utf8').trim(), 10);

let output = "";

for (let i = 0; i <= Math.floor(n / 2); i++) {
    output += " ".repeat(Math.floor(n / 2) - i) + "*".repeat(2 * i + 1) + "\n";
}

for (let i = Math.floor(n / 2) - 1; i >= 0; i--) {
    output += " ".repeat(Math.floor(n / 2) - i) + "*".repeat(2 * i + 1) + "\n";
}

// Write to output.txt
fs.writeFileSync('/home/abinav/git/amfoss-tasks/task-03/sub-task 4/Javascript/output.txt', output);

File: task-03/sub-task 4/Java/code.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class DiamondPattern {
    public static void main(String[] args) throws IOException {

        Scanner scanner = new Scanner(new File("/home/abinav/git/amfoss-tasks/task-03/sub-task 4/input.txt"));
        int n = scanner.nextInt();
        scanner.close();

        FileWriter writer = new FileWriter("/home/abinav/git/amfoss-tasks/task-03/sub-task 4/Java/output.txt");

        for (int i = 0; i <= n / 2; i++) {
            writer.write(" ".repeat((n / 2) - i) + "*".repeat(2 * i + 1) + "\n");
        }
)
        for (int i = (n / 2) - 1; i >= 0; i--) {
            writer.write(" ".repeat((n / 2) - i) + "*".repeat(2 * i + 1) + "\n");
        }

        writer.close();
    }
}
